examples/api/lib/widgets/implicit_animations/animated_value.0.dart

```diff
@@ -0,0 +1,371 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:math' as math;
+
+import 'package:flutter/material.dart';
+
+/// Flutter code sample for [AnimatedValue].
+
+void main() => runApp(const AnimatedValueApp());
+
+/// Prizes come in all shapes!
+enum Prize implements ValueKey<Prize> {
+  circle,
+  triangle,
+  square,
+  hexagon;
+
+  @override
+  Prize get value => this;
+
+  @override
+  String toString() => name;
+}
+
+class PrizeSpinner extends StatelessWidget {
+  const PrizeSpinner({super.key});
+
+  // This build method returns an AnimatedRotation, which is a subtype
+  // of AnimatedValue!
+  @override
+  AnimatedValue<double> build(BuildContext context) {
+    final Prize? prize = FreePrizes.maybeOf(context);
+    final double turns = (prize?.index ?? math.Random().nextDouble()) / 4 + 1 / 8;
```

Comment:
There's also a lot of magic numbers in this code. Maybe use named constants or comments to explain what these all mean.

Reply:
Or, if they can be derived from e.g. Enum.values.length maybe do that?